nav2_route/src/graph_loader.cpp

```diff
@@ -80,24 +81,63 @@ bool GraphLoader::loadGraphFromFile(
     filepath = graph_filepath_;
   }
 
-  bool result = false;
   if (!graph_file_loader_->fileExists(filepath)) {
     RCLCPP_ERROR(logger_, "The filepath %s does not exist", filepath.c_str());
     return false;
   }
   RCLCPP_INFO(
     logger_,
     "Loading graph file from %s, by parser %s", filepath.c_str(), plugin_type_.c_str());
-  result = graph_file_loader_->loadGraphFromFile(graph, graph_to_id_map, filepath);
 
-  return result;
+  if (!graph_file_loader_->loadGraphFromFile(graph, graph_to_id_map, filepath)) {
+    return false;
+  }
+
+  if (!transformGraph(graph)){
+    return false;
+  }
 
-  // Validate file is legit using a plugin API TODO(sm)
-  // Load file using a plugin API
+  return true;
 
   // Convert all coordinates to `frame` (in a new method) for standardization
   // Including conversion of GPS coordinates, so we can populate it in some
   // cartesian frame necessary for traversal cost estimation and densifying
   // (and so we don't need to propogate it through our structures)
 }
+
+bool GraphLoader::transformGraph(Graph &graph)
+{
+  //TODO(jw): Can we assume one frame for all?
```

Comment:
Nope :frowning: 

What you can do though is store a `std::unordered_map<std::string, tf2::Transform>` so that you can lookup a transform from `my_frame` to `route_frame` and store it so we don't lookup for each node, only each **unique** node. Otherwise, we can use the cached transform from a previous node to transform the point

Reply:
So basically everything you have here is right, it just needs to be reshuffled and add that map caching. You don't need to store that in the class though - just locally in the function which is deleted out of scope is fine